posts/2022-04-15-libertydevops-gitjenkinshelmiks.adoc

```diff
@@ -0,0 +1,180 @@
+                         #eg: docker tag liberty-app:v1.0 docker.io/someid/liberty-app:v1.0
+                         docker push <remote-docker-image-repository-url>/<docker-repo-name>/liberty-$<code identifier>:$<docker image version>
+                         #eg: docker push docker.io/someid/liberty-app:v1.0  
+                       '''   
+                  }
+               }
+            }
+          }
+
+
+Following are the parameter that is used in this example code
+
+* Git token: Personal access token generated from github.
+* remote-docker-image-repository-url : Docker hub repository URL.
+* USERNAME/PASSWWORD: username and password to connect to docker registry.
+
+image::/img/blog/jenkins-cred-binding-and-corresponding-param.png[Pipeline credential binding and corresponding param,width=30%,align="center"]
+
+* code identifier: This is optional, a unique docker image identifier
+* docker image version: docker image version number, a unique identifier   
+
+*Security scan* can be done along with Maven build command (CLI commands way of application scanning) or can do it in a separate pipeline stage. 
+
+For static code analysis, SonarQube community edition can be used. Install SonarQube server by either using file start-up type from cli by downloading the package in link:https://www.sonarqube.org/success-download-community-edition/[SonarQube server installation package] or use docker way as explained in link:https://docs.sonarqube.org/latest/setup/get-started-2-minutes/[SonarQube server installation steps]. SonarQube jenkins client setup details are given at link:https://docs.sonarqube.org/latest/analysis/scan/sonarscanner-for-jenkins/[SonarQube client for jenkins]. 
+ 
+    CLI command example, integrated with Maven command: 
+    $ mvn package verify sonar:sonar -Dsonar.projectKey=sampleapp -Dsonar.host.url=http://localhost:9000 -Dsonar.login=7b13f240123456780479b13d09ba8
+   
+For Docker image scan, you can use link:https://hub.docker.com/r/aquasec/trivy/[Docker image scan with trivy]. This scan provides the vulnerability details of open source jars that are used in the application. Sample commands are given below
+
+       sh '''
+             docker run aquasec/trivy image $docker_login/sampleapp:v1.0
+             #eg: docker run aquasec/trivy image docker.io/someid/liberty-app:v1.0
+       '''
+
+
+== Deploying Jakarta EE & MicroProfile applications with Open Liberty to Kubernetes with Jenkins
+
+For simplicity, I will use the command line (CLI) option to configure Jenkins to deploy a Jakarta EE & MicroProfile application with Open Liberty to Kubernetes. You could use one of many tools though, such as Helm, Travis CI, Circle CI, etc.
+
+Create a new stage in the pipeline code and write all the needed commands between the shell option (sample commands give below):
```

Comment:
Replace with "In your pipeline code, add the CLI commands in a new stage. The following sample pipeline code does xxxx and xxxx." (explain what will happen if they use the sample code)

Reply:
Modified to "In your pipeline code, add these CLI commands in a new stage. The following sample pipeline code connects to IBM cloud from CLI and then connects to kubernetes cluster running inside that, further it runs all the kubernetes deployment related configurations ."